cli/view_refresh.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package cli
+
+import (
+	"github.com/sourcenetwork/immutable"
+	"github.com/spf13/cobra"
+
+	"github.com/sourcenetwork/defradb/client"
+)
+
+func MakeViewRefreshCommand() *cobra.Command {
+	var name string
+	var schemaRoot string
+	var versionID string
+	var getInactive bool
+	var cmd = &cobra.Command{
+		Use:   "refresh",
+		Short: "Refresh views.",
+		Long: `Refresh views, executing the underlying query and LensVm transforms and
+persisting the results.
+
+View is refreshed as the current user, meaning results returned for all subsequent query requests
+to the view will receive items generated using the user refreshing the view's permissions.
```

Comment:
thought: This is better but still not very clear. I assume a new user looking at this would still get confused.

suggestion: I said this at some point previously but I can't remember where exactly and John just reminded me of it: We should probably materialize views from the view point of the node and not that of a given user. Queries on the view can then be filtered through ACP for the user doing the request. This will alleviate the need to have multiple materialized versions of the view.

Reply:
> This will alleviate the need to have multiple materialized versions of the view.

This is an alternative, and long term we probably want both anyway (e.g. here's my personal quick view, go and have a look).

I just picked the easiest, as there is time pressure on this PR and getting the node user is a bit more involved.

> This is better but still not very clear. I assume a new user looking at this would still get confused

Suggestions appreciated, it is a moderately complex thing to state in a line or two.

Reply:
> Lets clear up the question of what user is used to generate the view and then we can improve the documentation. 

That sounds like unwanted scope creep forced into a time-sensitive and deliberately lean initial PR, but sure.

Reply:
Discussed in standup, user-identity is fine for now, long term we may want something else

Reply:
Discussed in the standup. This implementation is working for the use case that triggered this PR and as such we can leave it as is.

As for the wording of the documentation, here is an possible alternative:
```suggestion
View is refreshed as the current user, meaning the cached items will reflect that user's 
permissions. Subsequent query requests to the view, regardless of user, will receive 
items from that cache.
```

Reply:
Will update, thanks Fred

- [ ] Update doc